packages/examples/fortune-v2/reputation-oracle/server/src/modules/webhook/webhook.service.ts

```diff
@@ -0,0 +1,206 @@
+      } else {
+        this.reputationService.decreaseReputation(clientParams.network.chainId, recordingOracleAddress, ReputationEntityType.RECORDING_ORACLE);
+      }
+
+      await this.webhookRepository
+        .updateOne({
+          id: webhookEntity.id,
+        }, { status: WebhookStatus.COMPLETED })
+
+      return true;
+    } catch (e) {
+      if (webhookEntity.retriesCount >= RETRIES_COUNT_THRESHOLD) {
+        await this.webhookRepository
+          .updateOne({
+            id: webhookEntity.id,
+          }, { status: WebhookStatus.FAILED })
+      } else {
+        await this.webhookRepository
+          .updateOne({
+            id: webhookEntity.id,
+          }, { retriesCount: webhookEntity.retriesCount + 1, waitUntil: new Date() })
+      }
+
+      this.logger.log("An error occurred during webhook validation: ", e, WebhookService.name)
+
+      return false;
+    }
+  }
+
+  public async validateFortune(webhookEntity: WebhookIncomingEntity, manifest: ManifestDto): Promise<boolean> {  
+    const signer = this.ethersSigner.createWallet(this.configService.get<string>("WEB3_JOB_LAUNCHER_PRIVATE_KEY", "web3 private key"));
+    const clientParams = await InitClient.getParams(signer);
+    const escrowClient = new EscrowClient(clientParams)
+      
+    try {
+      const recordingOracleResultsUrl = await escrowClient.getResultsUrl(webhookEntity.escrowAddress);
+      const recordingOracleResults: FinalResult[] = await StorageClient.downloadFileFromUrl(recordingOracleResultsUrl).catch(() => []);
+
+      if (recordingOracleResults.length === 0) {
+        this.logger.log(ErrorResults.NoRecordingOracleResultsFound, WebhookService.name);
+        throw new Error(ErrorResults.NoRecordingOracleResultsFound);
+      }
+
+      const finalResults: FinalResult[] = recordingOracleResults.filter(item => checkCurseWords(item.solution) || !finalResults.some(result => result.solution === item.solution));
+      if (finalResults.length === 0) {
+        this.logger.log(ErrorResults.NoResultsHaveBeenVerified, WebhookService.name);
+        throw new Error(ErrorResults.NoResultsHaveBeenVerified);
+      }
+
+      const [uploadedResult] = await this.storageClient.uploadFiles([finalResults], this.bucket);
+
+      const finalResultsUrl = uploadedResult.key;
+      const finalResultsHash = uploadedResult.hash
+
+      await escrowClient.storeResults(webhookEntity.escrowAddress, finalResultsUrl, finalResultsHash);
+
+      const checkPassed = recordingOracleResults.length <= finalResults.length;
+
+      await this.webhookRepository
+        .updateOne({
+          id: webhookEntity.id,
+        }, { 
+          resultsUrl: finalResultsUrl,
+          checkPassed,
+          status: WebhookStatus.PAID,
+          retriesCount: 0
+        })
+      
+      const recipients = finalResults.map(item => item.workerAddress);
+      const amounts = new Array(recipients.length).fill(BigNumber.from(manifest.fundAmount).div(recipients.length));
```

Comment:
Please, take a look here, we are trying to get the payout amount for each worker from the fund amount, but now it includes fee amount (fundAmount + fee), I suggest separating these 2 fields and storing them separately in the manifest. Or other suggestions? @portuu3 

https://github.com/humanprotocol/human-protocol/pull/520/files#r1228372175